Sources/MosaiqueAssetsPicker/ViewControllers/AssetDetailViewController.swift

```diff
@@ -130,14 +143,17 @@ public final class AssetDetailViewController: UIViewController {
         }
         NotificationCenter.assetPicker.post(name: PhotoPickerPickAssetsNotificationName, object: downloads)
     }
-
-    @objc func resetSelection() {
-        collectionView.indexPathsForSelectedItems?.forEach { collectionView.deselectItem(at: $0, animated: true) }
-        viewModel.reset(withAssetCollection: viewModel.assetCollection)
-        navigationItem.leftBarButtonItem = nil
-        navigationItem.rightBarButtonItem?.isEnabled = false
-        navigationItem.hidesBackButton = false
+    
+    func updateNavigationItems() {
+        
+        guard let doneButton = navigationItem.rightBarButtonItem else { return }
+        
+        let selectedCount = (collectionView.indexPathsForSelectedItems ?? []).count
+        doneButton.isEnabled = selectedCount > 0
+        let suffix = (doneButton.isEnabled && configuration.selectionMode.case != .single) ? "(\(selectedCount))" : ""
```

Comment:
Altho it's a good idea, I think this should use a proper localisation mechanism, but maybe this is overkill ? At least adding a space would be nice.

Reply:
Inspired myself with another library on this, but we could provide more customisability indeed